smart_contract/src/rln_sc_test.rs

```diff
@@ -0,0 +1,323 @@
+                        match pending_tx.get_receipt().await {
+                            Ok(receipt) => {
+                                if receipt.status() {
+                                    println!("Registration successful!");
+                                    println!(
+                                        "   Transaction hash: 0x{}",
+                                        hex::encode(receipt.transaction_hash)
+                                    );
+                                    println!("   Block number: {:?}", receipt.block_number);
+                                    println!("   Gas used: {}", receipt.gas_used);
+
+                                    // Test 7: Verify registration was successful
+                                    match rln_contract
+                                        .members(test_identity_commitment)
+                                        .call()
+                                        .await
+                                    {
+                                        Ok(verified_member) => {
+                                            if verified_member.userAddress != Address::ZERO {
+                                                println!("Registration verified:");
+                                                println!(
+                                                    "   User: {}",
+                                                    verified_member.userAddress
+                                                );
+                                                println!("   Index: {}", verified_member.index);
+                                            } else {
+                                                println!("Registration not found in verification");
+                                            }
+                                        }
+                                        Err(e) => {
+                                            eprintln!("Failed to verify registration: {}", e);
+                                        }
+                                    }
+                                } else {
+                                    println!("Registration transaction failed");
+                                }
+                            }
+                            Err(e) => eprintln!("Failed to get receipt: {}", e),
+                        }
+                    }
+                    Err(e) => eprintln!("Failed to send registration transaction: {}", e),
+                }
+            } else {
+                println!("Member {} already registered:", test_identity_commitment);
+                println!("   User: {}", member.userAddress);
+                println!("   Index: {}", member.index);
+            }
+        }
+        Err(e) => {
+            eprintln!("Failed to check if member exists: {}", e);
+        }
+    }
+
+    // Test 8: Slash a member
+    match rln_contract.members(test_identity_commitment).call().await {
```

Comment:
No need to test this. Prover does not do "slashing". Slashing should have been tested by smart contract team